src/replay/execution.zig

```diff
@@ -443,18 +445,73 @@ pub fn processReplayResults(
     return processed_a_slot;
 }
 
+/// Analogous to [mark_dead_slot](https://github.com/anza-xyz/agave/blob/15635be1503566820331cd2c845675641a42d405/core/src/replay_stage.rs#L2255)
 fn markDeadSlot(
+    replay_state: ReplayExecutionState,
     dead_slot: Slot,
-    progress_map: *ProgressMap,
-    ledger_result_writer: *sig.ledger.LedgerResultWriter,
+    ancestor_hashes_replay_update_sender: *sig.sync.Channel(AncestorHashesReplayUpdate),
 ) !void {
     // TODO add getForkProgress
-    var fork_progress = progress_map.map.getPtr(dead_slot) orelse {
+    var fork_progress = replay_state.progress_map.map.getPtr(dead_slot) orelse {
         return error.MissingBankProgress;
     };
     fork_progress.is_dead = true;
-    try ledger_result_writer.setDeadSlot(dead_slot);
-    // TODO Add and update slot stats blockstore.slots_stats.mark_dead(slot);
+    try replay_state.ledger_result_writer.setDeadSlot(dead_slot);
+    // TODOs
+    // - blockstore.slots_stats.mark_dead(slot);
+    // - slot_status_notifier
+    // - rpc_subscriptions
+
+    const dead_state: replay.edge_cases.DeadState = .fromState(
+        .from(replay_state.logger),
+        dead_slot,
+        replay_state.duplicate_slots_tracker,
+        replay_state.duplicate_confirmed_slots,
+        replay_state.fork_choice,
+        replay_state.epoch_slots_frozen_slots,
+    );
+    try check_slot_agrees_with_cluster.dead(
+        replay_state.allocator,
+        .from(replay_state.logger),
+        dead_slot,
+        replay_state.slot_tracker.root,
+        replay_state.duplicate_slots_to_repair,
+        ancestor_hashes_replay_update_sender,
+        dead_state,
+    );
+
+    // If blockstore previously marked this slot as duplicate, invoke duplicate state as well
+    const maybe_duplicate_proof = try replay_state.ledger_reader.getDuplicateSlot(dead_slot);
+    defer if (maybe_duplicate_proof) |proof| {
+        replay_state.allocator.free(proof.shred1);
+        replay_state.allocator.free(proof.shred2);
```

Comment:
ideally getDuplicateSlot should take an allocator. with the current design, you need to use the ledger's allocator:
```suggestion
        replay_state.ledger_reader.allocator.free(proof.shred1);
        replay_state.ledger_reader.allocator.free(proof.shred2);
```